src/extras/render-passes/render-pass-depth-aware-blur.js

```diff
@@ -8,18 +11,87 @@ import { shaderChunks } from '../../scene/shader-lib/chunks/chunks.js';
+    /**
+     * Initializes a new blur render pass. It has to be called before the render pass can be used.
+     *
+     * @param {import('../../platform/graphics/render-target.js').RenderTarget|null} [renderTarget] - The render
+     * target to render into (output).
+     * @param {Object} options - Options for configuring the blur effect.
+     * @param {import('../../platform/graphics/texture.js').Texture} [options.resizeSource] - The source texture to be blurred.
+     * @param {number} [options.scaleX] - The scale factor for the render target width. Defaults to 1.
+     * @param {number} [options.scaleY] - The scale factor for the render target height. Defaults to 1.
+     * @param {number} [options.kernelSize] - The size of the blur kernel. Defaults to 7.
+     * @param {boolean} [options.depthAware] - Whether the blur should be depth-aware. Defaults to false.
+     * @param {BLUR_GAUSSIAN|import('../../scene/constants.js').BLUR_BOX} [options.type] - The type of blur to apply. Defaults to BLUR_GAUSSIAN.
+     * @param {Vec2} [options.direction] - The direction of the blur. Defaults to (1, 0).
+     * @param {string} [options.channels] - The color channels to apply the blur to ('r'|'g'|'b'|'a'|'rg'|..|'ba'|'rgb'|'gba'|'rgba'). Defaults to 'rgba'.
+     */
+    init(renderTarget = null, options = {}) {
+        super.init(renderTarget, options);
+
+        this.sourceTexture = options.resizeSource;
+
+        // FIXME: resizeSource has to be passed and it is not possible to define
+        // the default value for resizeSource.
+        // But the init function of RenderPass is defined as
+        // init(renderTarget = null, options) { ... }
+        // So, the option parameter has to be optional because it is passed after
+        // renderTarget that is an optional parameter.
+        if (this.sourceTexture === null) {
+            console.error('RenderPassDepthAwareBlur: resizeSource is not set in the options.');
+            return;
+        }
+
+        const kernelSize = options.kernelSize ?? 7;
+        const depthAware = options.depthAware ?? false;
+        const blurType = options.type ?? BLUR_GAUSSIAN;
+        const blurDirection = options.direction ?? new Vec2(1, 0);
+        const channels = options.channels ?? 'rgba';
+
+        let loop = '';
+
+        const weightCoefs =
+            blurType === BLUR_GAUSSIAN ?
+                math.gaussWeights(kernelSize) :
+                new Array(kernelSize).fill(1);
+
+        const kernelWidth = kernelSize / 2 | 0;
+        const middle = kernelWidth + (1 - kernelSize % 2);
+
+        loop += `
+                sum += texture2D(sourceTexture, uv0).${channels} * ${weightCoefs[middle].toFixed(4)};
+                totalWeight += ${weightCoefs[middle].toFixed(4)};`;
+
+        // TODO: move calculating UV coordinates to the vertex shader and pass them as varying
+        for (let i = 1; i <= kernelWidth; i++) {
```

Comment:
we're trying to move away from generating shader from javascript as much as possible, as it's a lot harder to understand and modify. We try to create a single shader string as much as possible, and then in code generate a list of defines to pre-pend. See `RenderPassCompose` as an example. It'd be great to modify this in a similar fashion.

Reply:
even though the shader here is trying to be pretty generic, which makes it harder (with the types and channels)